### PR TITLE
Check an exit buffer when leaving zone.

### PR DESCRIPTION
**Pull request by lliss**
Reduce false positives for zone change by creating an exit buffer of 1 meter
around the zone radius. Don't really leave the zone unless we leave the buffer
as well as the zone.

Connects #108

### CODE REVIEW

**Comment by lliss** [comment]
Difficult to test. You need to use chrome dev tools and manually move the location around so that you are dancing on the 1 meter boundary of a zone. The leave event shouldn't fire until you pass further than 1 meter out.

**Comment by caseypt** [comment]
@lliss do you have some test coordinates you could share?

File: src/js/questManager.js [hunk 1]
```diff
@@ -44,6 +45,29 @@ module.exports = {
...
+    } else if (_.isObject(oldValue) && _.isObject(newValue) && oldValue.id === newValue.id) {
+        // Same id
+        return true;
+    } else if (!_.isObject(oldValue) && _.isObject(newValue)) {
+        // Moving into a zone.
+        return false;
+    } else if (_.isObject(oldValue) && !_.isObject(newValue)) {
+        // Maybe leaving a zone, first check the exit buffer to ensure
+        // we don't leave because of a GPS glitch.
+        if (oldValue.latLng.distanceTo(oldValue.currentPosition) <= oldValue.radius + 1) {
```
Comment (caseypt) [inline src/js/questManager.js]: Consider pulling out this number to the top of the function.

File: src/js/questManager.js [hunk 2]
```diff
@@ -44,6 +45,29 @@ module.exports = {
     }
 };
 
+function exitBuffer(oldValue, newValue) {
+    if (oldValue === newValue) {
+       // same object
+       return true;
+    } else if (_.isObject(oldValue) && _.isObject(newValue) && oldValue.id === newValue.id) {
```
Comment (caseypt) [inline src/js/questManager.js]: What comes through that's not an object?

**Comment by caseypt** [comment]
+1. Appears to be working, though we should remember to verify this in the field.

File: src/js/questManager.js (hunk 2 above)
Comment (lliss) [inline src/js/questManager.js]: undefined.

File: src/js/questManager.js (hunk 1 above)
Comment (lliss) [inline src/js/questManager.js]: Good call. Done and rebased.

**Pull request merged**